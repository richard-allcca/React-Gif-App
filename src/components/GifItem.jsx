import React from 'react'
import PropTypes from 'prop-types'

const GifItem = ({ title, url }) => {
  return (
    <div className='card'>
      <img src={url} alt={title} />
      <p>{title || 'sin titulo'}</p>
    </div>
  )
}

//  AÃ±adir PropTypes
GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}



export default GifItem