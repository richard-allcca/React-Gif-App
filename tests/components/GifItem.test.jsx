import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"


describe('Pruebas en el componente <GifItem/>', () => {

  const title = 'Titulo del Gif'
  const url = 'https://media1.giphy.com/media/3o7btLqYqjCQQqQqZS/200w.gif'


  test('Debe retornar un snapshot', () => {

    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot()
  })

  test('should mostrar el URL y ALT de la <img>', () => {

    render(<GifItem title={title} url={url} />)

    //? Muestra el URL de la imagen y el ALT de la misma
    // expect(screen.getByRole('img').src).toContain(url)
    // expect(screen.getByRole('img').alt).toContain(title )

    // MÃ©todo mejorado
    const { src, alt } = screen.getByRole('img')

    expect(src).toContain(url);
    expect(alt).toContain(alt);
  })

  test('Debe mostrar el titulo en el componente', () => {

    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()

  })


})