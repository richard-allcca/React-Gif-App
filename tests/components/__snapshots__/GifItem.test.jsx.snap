// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifItem/> Debe retornar un snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo del Gif"
      src="https://media1.giphy.com/media/3o7btLqYqjCQQqQqZS/200w.gif"
    />
    <p>
      Titulo del Gif
    </p>
  </div>
</div>
`;
